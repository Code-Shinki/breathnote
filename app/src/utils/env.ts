export const SITE_DOMAIN = process.env.NEXT_PUBLIC_SITE_DOMAIN || ''
export const SITE_TITLE = process.env.NEXT_PUBLIC_SITE_TITLE || ''
export const SITE_SUBTITLE = process.env.NEXT_PUBLIC_SITE_SUBTITLE || ''
export const SITE_DESCRIPTION = process.env.NEXT_PUBLIC_SITE_DESCRIPTION || ''
export const ARTICLE_PER_PAGE = Number(process.env.NEXT_PUBLIC_ARTICLE_PER_PAGE) || 12
export const GA_TRACKING_ID = process.env.NEXT_PUBLIC_GA_TRACKING_ID || ''
export const REVALIDATE_TIME = Number(process.env.REVALIDATE_TIME) || 60

export const API_KEY = process.env.API_KEY || ''
export const API_ENDPOINT = process.env.API_ENDPOINT || ''
export const DRAFT_TOKEN = process.env.DRAFT_TOKEN || ''
