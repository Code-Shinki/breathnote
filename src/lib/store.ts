/* ＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/

  LIB -> STORE

  - microCMS から非同期処理で取得したデータを保存

  - CONFIG : サイト設定 (オブジェクト)
  - ALL_POSTS : 記事一覧 (配列)
  - ALL_CATEGORIES : カテゴリ一覧 (配列)
  - ALL_TAGS : タグ一覧 (配列)

＿/ ＿/ ＿/ ＿/ ＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/＿/ */

import { AllCategoriesType, AllConfigType, AllPostsType, AllTagsType } from 'lib/types'

export const CONFIG = (async () => {
  const key = {
    headers: { 'X-API-KEY': process.env.API_KEY as string },
  }
  const fetchData: AllConfigType = await fetch('https://breathnote.microcms.io/api/v1/settings/config', key)
    .then((res) => res.json())
    .catch((err) => console.error(err))

  return fetchData
})()

export const ALL_POSTS = (async () => {
  const key = {
    headers: { 'X-API-KEY': process.env.API_KEY as string },
  }
  const fetchData: AllPostsType = await fetch('https://breathnote.microcms.io/api/v1/posts', key)
    .then((res) => res.json())
    .catch((err) => console.error(err))

  return fetchData.contents
})()

export const ALL_CATEGORIES = (async () => {
  const key = {
    headers: { 'X-API-KEY': process.env.API_KEY as string },
  }
  const fetchData: AllCategoriesType = await fetch('https://breathnote.microcms.io/api/v1/categories', key)
    .then((res) => res.json())
    .catch((err) => console.error(err))

  return fetchData.contents
})()

export const ALL_TAGS = (async () => {
  const key = {
    headers: { 'X-API-KEY': process.env.API_KEY as string },
  }
  const fetchData: AllTagsType = await fetch('https://breathnote.microcms.io/api/v1/tags', key)
    .then((res) => res.json())
    .catch((err) => console.error(err))

  return fetchData.contents
})()
