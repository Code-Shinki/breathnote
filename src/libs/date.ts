import { format } from 'date-fns'
import { utcToZonedTime } from 'date-fns-tz'

export const getFormattedDate = (date: string | Date | null, conversion: string) => {
  // 下書き中は任意の文字列を返す
  if (!date) return 'Draft'

  const jstDate = utcToZonedTime(new Date(date), 'Asia/Tokyo')

  return format(jstDate, conversion)
}
