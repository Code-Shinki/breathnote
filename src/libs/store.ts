import axios from 'axios'
import { AllKeywords } from 'types/keyword'
import { AllPosts } from 'types/post'
import { API_ENDPOINT, API_KEY } from 'utils/env'

const KEY = { headers: { 'X-API-KEY': API_KEY } }

const fetchAllPosts = async () => {
  console.log('ALL_POSTS fire')

  // 一度に受け取る記事数
  const limit = 10

  // 総記事数等を含む最初の10記事を取得
  const posts: AllPosts = await axios
    .get(`${API_ENDPOINT}/posts?limit=${limit}`, KEY)
    .then((res) => res.data)
    .catch((err) => {
      throw new Error(`FETCH FAILED (fetchAllPosts() - First reqest) : ${err}`)
    })

  // APIを叩く回数
  const requestsCount = Math.floor(posts.totalCount / limit)

  // すべての記事を取得してposts.contentsに追加
  for (let i = 1; i <= requestsCount; i++) {
    const fetchData = await axios
      .get(`${API_ENDPOINT}/posts?offset=${limit * i}&limit=${limit}`, KEY)
      .then((res) => res.data)
      .catch((err) => {
        throw new Error(`FETCH FAILED (fetchAllPosts() - Roop ${i} reqest) : ${err}`)
      })
    posts.contents = posts.contents.concat(fetchData.contents)
  }

  // 記事を降順 (公開日が新しい順) に並び替え
  if (posts.contents.length > 1) {
    posts.contents.sort((a, b) => {
      if (a.publishedAt < b.publishedAt) {
        return 1
      } else {
        return -1
      }
    })
  }

  return posts
}

const fetchAllKeywords = async () => {
  console.log('ALL_KEYWORDS fire')

  // 一度に受け取るキーワード数
  const limit = 10

  // 総キーワード数等を含む最初の10データを取得
  const keywords: AllKeywords = await axios
    .get(`${API_ENDPOINT}/keywords?limit=${limit}`, KEY)
    .then((res) => res.data)
    .catch((err) => {
      throw new Error(`FETCH FAILED (fetchAllKeywords() - First reqest) : ${err}`)
    })

  // APIを叩く回数
  const requestsCount = Math.floor(keywords.totalCount / limit)

  // すべてのキーワードを取得してposts.contentsに追加
  for (let i = 1; i <= requestsCount; i++) {
    const fetchData = await axios
      .get(`${API_ENDPOINT}/keywords?offset=${limit * i}&limit=${limit}`, KEY)
      .then((res) => res.data)
      .catch((err) => {
        throw new Error(`FETCH FAILED (fetchAllKeywords() - Roop ${i} reqest) : ${err}`)
      })
    keywords.contents = keywords.contents.concat(fetchData.contents)
  }

  // キーワードを昇順 (アルファベット順) に並び替え
  if (keywords.contents.length > 1) {
    keywords.contents.sort((a, b) => {
      if (a.name > b.name) {
        return 1
      } else {
        return -1
      }
    })
  }

  return keywords
}

export const ALL_POSTS = fetchAllPosts()
export const ALL_KEYWORDS = fetchAllKeywords()
