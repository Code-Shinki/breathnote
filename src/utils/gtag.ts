import { GA_TRACKING_ID } from 'utils/env'

export const pageview = (url: string): void => {
  window.gtag('config', GA_TRACKING_ID, {
    page_path: url,
  })
}

type GaEventProps = {
  action: string
  category: string
  label: string
  value?: number
}

export const event = ({ action, category, label, value }: GaEventProps): void => {
  window.gtag('event', action, {
    event_category: category,
    event_label: label,
    value: value,
  })
}
