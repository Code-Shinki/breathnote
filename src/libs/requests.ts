import { ALL_POSTS } from 'libs/store'

// keywords/index.tsx, keywords/page/[offset].tsx
export const getKeywordPosts = async (slug: string) => {
  const posts = (await ALL_POSTS).contents
  const keywordPosts = posts.filter((post) => {
    return post.keywords.some((keyword) => {
      return keyword.slug === slug
    })
  })
  return keywordPosts
}
