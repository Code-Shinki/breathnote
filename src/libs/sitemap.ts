import { getFormattedDate } from 'libs/date'
import { Post } from 'types/post'
import { SITE_DOMAIN } from 'utils/env'

export const generateSitemap = (posts: Post[]) => {
  return `<?xml version="1.0" encoding="UTF-8"?>
            <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
              xmlns:xhtml="http://www.w3.org/1999/xhtml"
              xmlns:mobile="http://www.google.com/schemas/sitemap-mobile/1.0"
              xmlns:video="http://www.google.com/schemas/sitemap-video/1.1"
              xmlns:image="http://www.google.com/schemas/sitemap-image/1.1">
                <url>
                  <loc>${`${SITE_DOMAIN}/`}</loc>
                  <lastmod>${getFormattedDate(posts[0].publishedAt, 'yyyy-MM-dd')}</lastmod>
                  <changefreq>daily</changefreq>
                  <priority>1.0</priority>
                </url>
                <url>
                  <loc>${`${SITE_DOMAIN}/about`}</loc>
                  <changefreq>never</changefreq>
                </url>
              ${posts
                .map((post) => {
                  return `
                    <url>
                      <loc>${`${SITE_DOMAIN}/posts/${post.slug}`}</loc>
                      <lastmod>${getFormattedDate(post.publishedAt, 'yyyy-MM-dd')}</lastmod>
                      <changefreq>weekly</changefreq>
                    </url>
                  `
                })
                .join('')}
            </urlset>
  `
}
