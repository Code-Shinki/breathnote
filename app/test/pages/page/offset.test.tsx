import { cleanup, render } from '@testing-library/react'
import { fetchAllKeywords, fetchAllPosts } from '../../../src/libs/store'
import Home from '../../../src/pages/page/[offset]'
import { SITE_DESCRIPTION, SITE_DOMAIN, SITE_SUBTITLE, SITE_TITLE } from '../../../src/utils/env'

describe(`TOP PAGE 4/7 (pages/page/[offset].tsx)`, () => {
  afterEach(cleanup)

  test('snapshot', async () => {
    const posts = (await fetchAllPosts()).contents.slice(-5)
    const offset = 4
    const postLength = 84
    const pageProps = {
      url: `${SITE_DOMAIN}/page/${offset}`,
      type: 'home',
      title: `${SITE_TITLE} (${offset}) - ${SITE_SUBTITLE}`,
      description: SITE_DESCRIPTION,
      keywords: (await fetchAllKeywords()).contents,
    }

    const { asFragment } = render(<Home pageProps={pageProps} posts={posts} postLength={postLength} offset={offset} />)
    const tree = asFragment()
    expect(tree).toMatchSnapshot()
  })
})
