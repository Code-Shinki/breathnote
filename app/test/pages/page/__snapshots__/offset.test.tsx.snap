// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TOP PAGE 4/7 (pages/page/[offset].tsx) snapshot 1`] = `
<DocumentFragment>
  <section
    class="root"
  >
    <article
      class="article"
    >
      <div
        class="thumbnail"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27850%27%20height=%27445%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Material-UIとreact-springの併用でエラー（Cannot update a component ForwardRef Modal ...）"
            data-nimg="intrinsic"
            decoding="async"
            src="/_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2Ffb5d600012fa4390bb62fcae2998fcf2%2Fmaterial-ui.png&w=1200&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2Ffb5d600012fa4390bb62fcae2998fcf2%2Fmaterial-ui.png&w=1024&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2Ffb5d600012fa4390bb62fcae2998fcf2%2Fmaterial-ui.png&w=1200&q=75 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
      </div>
      <time
        class="date"
        datetime="2021-03-19T14:47:17.358Z"
        itemprop="datepublished"
      >
        2021年3月19日
      </time>
      <h2
        class="title"
      >
        <a
          href="/posts/react-spring-modal-state-error"
        >
          Material-UIとreact-springの併用でエラー（Cannot update a component ForwardRef Modal ...）
        </a>
      </h2>
      <hr />
      <ul
        class="keywords"
      >
        <li>
          React
        </li>
        <li>
          Material-UI
        </li>
      </ul>
      <div
        class="loader"
      >
        <div
          class="inner1"
        />
        <div
          class="inner2"
        />
        <div
          class="inner3"
        />
      </div>
    </article>
    <article
      class="article"
    >
      <div
        class="thumbnail"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27850%27%20height=%27445%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Reactではstateの更新が即時反映されない場合がある"
            data-nimg="intrinsic"
            decoding="async"
            src="/_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2F36945828ef544a2fb37367d845aeaf05%2Freact.png&w=1200&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2F36945828ef544a2fb37367d845aeaf05%2Freact.png&w=1024&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2F36945828ef544a2fb37367d845aeaf05%2Freact.png&w=1200&q=75 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
      </div>
      <time
        class="date"
        datetime="2021-03-16T16:08:28.810Z"
        itemprop="datepublished"
      >
        2021年3月17日
      </time>
      <h2
        class="title"
      >
        <a
          href="/posts/react-state-not-reflected"
        >
          Reactではstateの更新が即時反映されない場合がある
        </a>
      </h2>
      <hr />
      <ul
        class="keywords"
      >
        <li>
          React
        </li>
      </ul>
      <div
        class="loader"
      >
        <div
          class="inner1"
        />
        <div
          class="inner2"
        />
        <div
          class="inner3"
        />
      </div>
    </article>
    <article
      class="article"
    >
      <div
        class="thumbnail"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27850%27%20height=%27445%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Jestを用いたNext.jsのテストで環境変数を利用する"
            data-nimg="intrinsic"
            decoding="async"
            src="/_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2Fea80ec90e22d488f86164ab667c78201%2Fjest.png&w=1200&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2Fea80ec90e22d488f86164ab667c78201%2Fjest.png&w=1024&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2Fea80ec90e22d488f86164ab667c78201%2Fjest.png&w=1200&q=75 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
      </div>
      <time
        class="date"
        datetime="2021-03-05T06:00:02.829Z"
        itemprop="datepublished"
      >
        2021年3月5日
      </time>
      <h2
        class="title"
      >
        <a
          href="/posts/nextjs-jest-environment-variable"
        >
          Jestを用いたNext.jsのテストで環境変数を利用する
        </a>
      </h2>
      <hr />
      <ul
        class="keywords"
      >
        <li>
          Next.js
        </li>
        <li>
          Jest
        </li>
      </ul>
      <div
        class="loader"
      >
        <div
          class="inner1"
        />
        <div
          class="inner2"
        />
        <div
          class="inner3"
        />
      </div>
    </article>
    <article
      class="article"
    >
      <div
        class="thumbnail"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27850%27%20height=%27445%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="ヘッドレスCMSから受け取ったMDXをNext.jsコンポーネントに変換する"
            data-nimg="intrinsic"
            decoding="async"
            src="/_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2F8459896f7b564d7faee484bc9124e62b%2Fnext-js.png&w=1200&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2F8459896f7b564d7faee484bc9124e62b%2Fnext-js.png&w=1024&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2F8459896f7b564d7faee484bc9124e62b%2Fnext-js.png&w=1200&q=75 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
      </div>
      <time
        class="date"
        datetime="2021-02-27T06:00:02.752Z"
        itemprop="datepublished"
      >
        2021年2月27日
      </time>
      <h2
        class="title"
      >
        <a
          href="/posts/nextjs-markdown-to-jsx"
        >
          ヘッドレスCMSから受け取ったMDXをNext.jsコンポーネントに変換する
        </a>
      </h2>
      <hr />
      <ul
        class="keywords"
      >
        <li>
          Next.js
        </li>
        <li>
          Jamstack
        </li>
      </ul>
      <div
        class="loader"
      >
        <div
          class="inner1"
        />
        <div
          class="inner2"
        />
        <div
          class="inner3"
        />
      </div>
    </article>
    <article
      class="article"
    >
      <div
        class="thumbnail"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27850%27%20height=%27445%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="JamstackなNext.jsブログにプレビューモードを実装する"
            data-nimg="intrinsic"
            decoding="async"
            src="/_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2F8459896f7b564d7faee484bc9124e62b%2Fnext-js.png&w=1200&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2F8459896f7b564d7faee484bc9124e62b%2Fnext-js.png&w=1024&q=75 1x, /_next/image?url=https%3A%2F%2Fimages.microcms-assets.io%2Fassets%2F3c39a709d91a45e381e9d2d9c424a09b%2F8459896f7b564d7faee484bc9124e62b%2Fnext-js.png&w=1200&q=75 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
      </div>
      <time
        class="date"
        datetime="2021-02-23T14:16:00.797Z"
        itemprop="datepublished"
      >
        2021年2月23日
      </time>
      <h2
        class="title"
      >
        <a
          href="/posts/nextjs-preview-mode"
        >
          JamstackなNext.jsブログにプレビューモードを実装する
        </a>
      </h2>
      <hr />
      <ul
        class="keywords"
      >
        <li>
          Next.js
        </li>
        <li>
          Jamstack
        </li>
      </ul>
      <div
        class="loader"
      >
        <div
          class="inner1"
        />
        <div
          class="inner2"
        />
        <div
          class="inner3"
        />
      </div>
    </article>
  </section>
  <nav
    aria-label="pagination"
    class="nav"
  >
    <ul
      class="ul"
    >
      <li
        class="li false"
      >
        <a
          class="link"
          href="/"
        >
          1
        </a>
      </li>
      <li
        class="dotted"
      >
        ...
      </li>
      <li
        class="li false"
      >
        <a
          class="link"
          href="/page/3"
        >
          3
        </a>
      </li>
      <li
        class="li li-current"
      >
        <a
          class="link"
          href="/page/4"
        >
          4
        </a>
      </li>
      <li
        class="li false"
      >
        <a
          class="link"
          href="/page/5"
        >
          5
        </a>
      </li>
      <li
        class="dotted"
      >
        ...
      </li>
      <li
        class="li false"
      >
        <a
          class="link"
          href="/page/7"
        >
          7
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
